erpnext/hr/doctype/expense_receipt/expense_receipt.json

```diff
@@ -0,0 +1,536 @@
+{
```

Comment:
I've mostly adapted it to what you've suggested here. I shifted the expense type to the right side because it balances out the form better but still keeps all the required information at the top.